osmoutils/cache_ctx.go

```diff
@@ -35,6 +43,19 @@ func ApplyFuncIfNoError(ctx sdk.Context, f func(ctx sdk.Context) error) (err err
 	return err
 }
 
+// Frustratingly, this has to return the error descriptor, not an actual error itself
+// because the SDK errors here are not actually errors. (They don't implement error interface)
+func IsOutOfGasError(err any) (bool, string) {
+	switch e := err.(type) {
+	case types.ErrorOutOfGas:
+		return true, e.Descriptor
+	case types.ErrorGasOverflow:
+		return true, e.Descriptor
```

Comment:
What about types.ErrorNegativeGasConsumed?

Reply:
oh wow didn't know that existed, let me think about it

Reply:
hmmmmm, if only those were sum types and we could ensure the the whole space is covered... 🤔 (rust)